crates/fraud_proofs/global_merkle_root/storage/src/update.rs

```diff
@@ -305,6 +307,265 @@ impl TransactionOutputs for Transaction {
+            .storage_as_ref::<ContractsLatestUtxo>()
+            .get(&contract_id)
+            .unwrap()
+            .unwrap()
+            .into_owned();
+
+        // Then
+        assert_eq!(inserted_contract_utxo.utxo_id(), &utxo_id);
+    }
+
+    #[test]
+    /// When encountering a transaction with a contract output,
+    /// `process_output` should ensure an appropriate contract UTxO is
+    /// populated in the `ContractCreated` table.
+    fn process_output__should_update_latest_contract_utxo_when_interacting_with_contract()
+    {
+        let mut rng = StdRng::seed_from_u64(1337);
+
+        // Given
+        let mut storage: InMemoryStorage<Column> = InMemoryStorage::default();
+        let mut storage_tx = storage.write_transaction();
+        let mut storage_update_tx =
+            storage_tx.construct_update_merkleized_tables_transaction();
+
+        let tx_pointer = random_tx_pointer(&mut rng);
+        let utxo_id = random_utxo_id(&mut rng);
+
+        let contract_id = random_contract_id(&mut rng);
+        let input_contract = input::contract::Contract {
+            contract_id,
+            ..Default::default()
+        };
+        let inputs = vec![Input::Contract(input_contract)];
+
+        let output_contract = output::contract::Contract {
+            input_index: 0,
+            ..Default::default()
+        };
+
+        let output = Output::Contract(output_contract);
+
+        // When
+        storage_update_tx
+            .process_output(tx_pointer, utxo_id, &inputs, &output)
+            .unwrap();
+
+        storage_tx.commit().unwrap();
+
+        let inserted_contract_utxo = storage
+            .read_transaction()
+            .storage_as_ref::<ContractsLatestUtxo>()
+            .get(&contract_id)
+            .unwrap()
+            .unwrap()
+            .into_owned();
+
+        // Then
+        assert_eq!(inserted_contract_utxo.utxo_id(), &utxo_id);
+    }
+
+    #[test]
+    /// When encountering a transaction with a coin input,
+    /// `process_input` should ensure this coin is
+    /// removed from the `Coins` table, as this coin is no longer
+    /// a part of the active UTxO set.
+    fn process_input__should_remove_coin() {
+        let mut rng = StdRng::seed_from_u64(1337);
+
+        // Given
+        let mut storage: InMemoryStorage<Column> = InMemoryStorage::default();
+        let mut storage_tx = storage.write_transaction();
+        let mut storage_update_tx =
+            storage_tx.construct_update_merkleized_tables_transaction();
+
+        let output_amount = rng.gen();
+        let output_address = random_address(&mut rng);
+        let tx_pointer = random_tx_pointer(&mut rng);
+        let utxo_id = random_utxo_id(&mut rng);
+        let inputs = vec![];
+
+        let output = Output::Coin {
+            to: output_address,
+            amount: output_amount,
+            asset_id: AssetId::zeroed(),
+        };
+
+        let input = Input::CoinSigned(CoinSigned {
```

Comment:
I don't see where you inserted this coin. If you do it somewhere already, then you would be nice if you checked here that coin exists in the `Coins` table=)

Reply:
It's being inserted in `storage_update_tx.process_output`. I can add an assertion to make this more clear, thanks for highlighting.

Reply:
Done in https://github.com/FuelLabs/fuel-core/pull/2598/commits/2457f06e8390914ee9a2530d3e71441f143b3935